test/dhtest.c
=======================================================
lhs: 100644 | 5940aa78b2efea674e804104a81ddd9d2ba5cd7b
rhs: 100644 | 8fad6da4fd19e51b91dae6f889b84ea41b022c22
---@@ -194,6 +194,7 @@ int main(int argc, char *argv[])
     if (!run_rfc5114_tests())
         ret = 1;
  err:
+    (void)BIO_flush(out);
     ERR_print_errors_fp(stderr);
 
     OPENSSL_free(abuf);
@@ -202,6 +203,12 @@ int main(int argc, char *argv[])
     DH_free(a);
     BN_GENCB_free(_cb);
     BIO_free(out);
+
+#ifndef OPENSSL_NO_CRYPTO_MDEBUG
+    if (CRYPTO_mem_leaks_fp(stderr) <= 0)
+        ret = 1;
+#endif
+
     EXIT(ret);
 }
 

---